### PR TITLE
Moved some functions out of lib/analytics.js

### PR DESCRIPTION
**Pull request by navneetsaluja**
Works on #4474 

refactor

### Technical

To be edited after final review. 

### Stakeholders

@jdlrobson

### CODE REVIEW

**Comment by navneetsaluja** [comment]
@/jdlrobson 
We already have ol.analytics.js, do we need to create another `analytics.js` file or can we add functions from the html to this file only? 

Also, is there a way to access python functions in .js files? I need `$total_time` in the JS script but I cannot figure out a way to access it from `ol.analytics.js`. In the meantime I've transferred rest of the functions to `ol.analytics.js`

**Comment by jdlrobson** [comment]
You can add to the existing JS file

On Wed, Mar 17, 2021, 12:57 PM Navneet Saluja ***@***.***>
wrote:

> @/jdlrobson
> We already have ol.analytics.js, do we need to create another analytics.js
> file or can we add functions from the html to this file only?
>
> Also, is there a way to access python functions in .js files? I need
> $total_time in the JS script but I cannot figure out a way to access it
> from ol.analytics.js. In the meantime I've transferred rest of the
> functions to ol.analytics.js
>

You'll need to make use of a data- attribute. There are a few examples if
you search for `data-` in the codebase.

**Comment by navneetsaluja** [comment]
@jdlrobson Updated, please take a look. 

Only problem remaining is making `vs` the variable, accessible as a global variable. Earlier, when we had `analytics.html` we could do following: Go to https://openlibrary.org/, open the console, enter vs and see the array with the statistics. After moving the code, vs is not accessible from the console anymore. Could you help me with this?

Also, the `npm run test` fails because ` Jest: "global" coverage threshold for functions (19%) not met: 18.96%`, what does this mean?

**Comment by jdlrobson** [comment]
You can register a global variable using  `window.vs = vs`

It is asking for you to either lowering the thresholds in package.json or by adding a unit test for the code you have moved.

**Comment by navneetsaluja** [comment]
@jdlrobson Weird how it didn't give an error when I locally ran `npm run test` but when I committed, it gave the threshold error, is this intended? I did modify the threshold if that's alright. Please take a look, thank you.

File: openlibrary/plugins/openlibrary/js/ol.analytics.js [hunk 1]
```diff
@@ -23,5 +25,26 @@ export default function initAnalytics() {
...
+        for (i in vs) {
+            vs[i]['cache_bust']=Math.random();
+            vs[i]['server_ms']=$('.analytics-stats-time-calculator').data('time');
+            vs[i]['server_name']='ol-web.us.archive.org';
+            vs[i]['service']='ol';
+        }
+        if (window.flights){
+            window.flights.init();
+        }
+        if ($('.more_search').size()>0) {
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/ol.analytics.js:39]: Note, size is a deprecated jquery function. We should make sure to use length in a follow up.

**Review by jdlrobson — APPROVED** [review]
Looks good. I just need to test this and verify the code still works.

File: openlibrary/plugins/openlibrary/js/ol.analytics.js (hunk 1 above)
Comment (navneetsaluja) [reply openlibrary/plugins/openlibrary/js/ol.analytics.js:39]: Right, noted. 
As written in my first comment, I'm still confused thought, there is no mention of a `more_search` class or `window.flights` anywhere else in the code, could you verify? Maybe it is redundant code?

Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/ol.analytics.js:39]: It's very possible. I'm out and about right now but when i have my computer I can take a closer look. Thanks for flagging this potentially dead code. That's really helpful!

**Comment by jdlrobson** [comment]
Tested locally! Sorry for the delay here and thank you for the contribution!

File: openlibrary/plugins/openlibrary/js/ol.analytics.js (hunk 1 above)
Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/ol.analytics.js:39]: I am pretty sure you are right, the more_search selector  matches nothing. If you could  submit a follow up patch to remove this line, that would be super helpful!

Comment (navneetsaluja) [reply openlibrary/plugins/openlibrary/js/ol.analytics.js:39]: Gotcha, opening a PR soon.

**Pull request merged**